apps/ai-image-tagging/frontend/src/components/AITagView/AITagView.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { CheckboxField, TextInput, Pill, Button } from '@contentful/forma-36-react-components';
+
+import { styles } from './styles';
+
+async function callAPI(url) {
+  const data = await fetch(`http://localhot:3000/tags/${url}`).then(res => res.json());
+  return data.tags;
+}
+
+export class AITagView extends React.Component {
+
+  constructor(props){
+    super(props);
+
+    this.state = {
+      value: "",
+      checked: false,
+      tags: props.entries.imageTags.getValue() || [],
+      overwrite: true,
+      isMissingImage: !props.entries.image.getValue(),
+      isFetchingTags: false
+    }
+  }
+
+  componentDidMount() {
+    this.props.entries.image.onValueChanged(() => {
+      this.setState(() => ({
+        isMissingImage: !this.props.entries.image.getValue()
+      }))
+    });
+  }
+
+  toggleOverwrite = () => {
+    this.setState(state => ({
+      overwrite: !state.overwrite
+    }));
+  }
+
+  addTag = async (e) => {
+    if (e.key !== "Enter") { return; }
+
+    const newTags = [e.target.value, ...this.state.tags];
+    try {
+      await this.props.entries.imageTags.setValue(newTags);
+      this.setState({
+        tags: newTags,
+        value: ""
+      });
+    } catch(e) {
+    }
+  }
+
+  deleteTag = async (tag) => {
+    const newTags = this.state.tags.filter(t => t !== tag);
+    try {
+      await this.props.entries.imageTags.setValue(newTags);
+      this.setState({
+        tags: newTags
+      });
+    } catch(e) {
+    }
+  }
+
+  fetchTags =  async () => {
+    const file = await this.props.space.getAsset(this.props.entries.image.getValue().sys.id);
+    const url = file.fields.file['en-US'].url.replace(/^\/\/images.ctfassets.net\//, '');
```

Comment:
You cannot assume they've got `en-US`. To get the default locale code use `locales.default` as described in https://www.contentful.com/developers/docs/extensibility/ui-extensions/sdk-reference/#locales

Reply:
Instead of using `replace` I'd parse the URL w/ https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

Reply:
I'd also use something like `lodash.get` to navigate these deeply nested data structures